### PR TITLE
[3.x] Initialize class variables with default values

### PR DESCRIPTION
**Pull request by qarmin**
Part of #43636 for 3.x branch.

It should decrease number of Cppcheck errors from 791 to ~125(except buffers and some rendering things).

It should be really easy to check this commit, because it adds only to constructor values initializations(in opposite to similar commits to master branch when I moved a lot of things across code and also initialized structs members which was not shown in Cppcheck).

### CODE REVIEW

File: modules/bullet/generic_6dof_joint_bullet.cpp [hunk 1]
```diff
@@ -43,6 +43,12 @@
 Generic6DOFJointBullet::Generic6DOFJointBullet(RigidBodyBullet *rbA, RigidBodyBullet *rbB, const Transform &frameInA, const Transform &frameInB) :
 		JointBullet() {
 
+	for (int i = 0; i < 3; i++) {
```
Comment (pouleyKetchoupp) [inline modules/bullet/generic_6dof_joint_bullet.cpp:46]: This has been fixed with a different code which is a bit cleaner on master:
```
for (int i = 0; i < 3; i++) {
	for (int j = 0; j < PhysicsServer3D::G6DOF_JOINT_FLAG_MAX; j++) {
		flags[i][j] = false;
	}
}
```

Also, I'm wondering if `memset` would be better in this case. I don't know if all compilers would optimize this loop properly.

**Comment by akien-mga** [comment]
That's a huge PR with a lot of surface for potential bugs. I'll try to review it soon but it takes time.

To fix #35877, it would be better to make a dedicated PR with only that fix.

File: core/io/resource_format_binary.cpp [hunk 2]
```diff
@@ -980,6 +980,8 @@ ResourceInteractiveLoaderBinary::ResourceInteractiveLoaderBinary() :
 		f(NULL),
 		error(OK),
 		stage(0) {
+	ver_format = 0;
+	importmd_ofs = 0;
```
Comment (aaronfranke) [inline core/io/resource_format_binary.cpp:984]: Why are some of these before the `{` and some after it?

**Review by aaronfranke — COMMENTED** [review]
I assume this is still desired, but it needs to be rebased on the latest 3.x branch. Also note that #35877 has been fixed via #45803.

File: core/io/resource_format_binary.cpp (hunk 2 above)
Comment (qarmin) [reply core/io/resource_format_binary.cpp:984]: Initializer list(before {) require to have all variables in same order as in class definition, so for me it was easier to write such code(in master initialization was moved from constructor)

**Comment by qarmin** [comment]
Currently I don't have enough time to rebase such commits, so feel free to use them to create new PR.

**Pull request closed without merging**